UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Filters/BespokeExchangeSetAuthorizationFilterAttribute.cs

```diff
@@ -17,18 +21,29 @@ public class BespokeExchangeSetAuthorizationFilterAttribute : ActionFilterAttrib
     {
         private const string TokenAudience = "aud";
         private const string ExchangeSetStandard = "exchangeSetStandard";
+        private const string TokenIssuer = "iss";
+        private const string TokenTenantId = "http://schemas.microsoft.com/identity/claims/tenantid";
         private readonly IOptions<AzureADConfiguration> azureAdConfiguration;
         private static readonly string[] ExchangeSetStandards = Enum.GetNames(typeof(ExchangeSetStandard));
-
-        public BespokeExchangeSetAuthorizationFilterAttribute(IOptions<AzureADConfiguration> azureAdConfiguration)
+        private readonly IConfiguration configuration;
+        private readonly IAzureAdB2CHelper azureAdB2CHelper;
+        public BespokeExchangeSetAuthorizationFilterAttribute(IOptions<AzureADConfiguration> azureAdConfiguration, IConfiguration configuration, IAzureAdB2CHelper azureAdB2CHelper)
         {
             this.azureAdConfiguration = azureAdConfiguration ?? throw new ArgumentNullException(nameof(azureAdConfiguration));
+            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            this.azureAdB2CHelper = azureAdB2CHelper ?? throw new ArgumentNullException(nameof(azureAdB2CHelper));
         }
 
         public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
         {
-            var tokenAudience = context.HttpContext.User.FindFirstValue(TokenAudience);
             var azureAdClientId = azureAdConfiguration.Value.ClientId;
+            var azureAdTenantId = azureAdConfiguration.Value.TenantId;
+
+            var tokenAudience = context.HttpContext.User.FindFirstValue(TokenAudience);
+            var tokenTenantId = context.HttpContext.User.FindFirstValue(TokenTenantId);
+            var tokenIssuer = context.HttpContext.User.FindFirstValue(TokenIssuer);
+            var correlationId = context.HttpContext.Request.Headers[CorrelationIdMiddleware.XCorrelationIdHeaderKey].FirstOrDefault();
+            correlationId = Guid.TryParse(correlationId, out Guid correlationIdGuid) ? correlationIdGuid.ToString() : Guid.Empty.ToString();
```

Comment:
If the Guid.TryParse fails, this would mean that the correlation Id in the header was invalid and we would probably want to log this.